File: openapi-test-kotlin-server-quarkus-reactive-client-mp-reactive/testapi/api/AnimalsApiClient.kt
package testapi.api

import io.smallrye.mutiny.Uni
import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import jakarta.ws.rs.Produces
import jakarta.ws.rs.WebApplicationException
import jakarta.ws.rs.core.GenericType
import jakarta.ws.rs.core.MediaType
import jakarta.ws.rs.core.Response
import java.lang.IllegalStateException
import kotlin.collections.List
import org.eclipse.microprofile.rest.client.inject.RegisterRestClient
import testapi.api.ListAnimalsResponse.Status200
import testapi.api.ListAnimalsResponse.Status4XX
import testapi.api.ListAnimalsResponse.Status5XX
import testapi.model.Animal
import testapi.model.Error

@RegisterRestClient
@Path("/animals")
sealed interface AnimalsApiClient : AnimalsApi {
  /** List all animals (polymorphic) - handles response status codes */
  override fun listAnimals(): Uni<ListAnimalsResponse> = listAnimalsRaw().map({ response: Response -> if (response.getStatus() == 200) { Status200(response.readEntity(object : GenericType<List<Animal>>() {})) }
  else if (response.getStatus() >= 400 && response.getStatus() < 500) { Status4XX(response.getStatus(), response.readEntity(Error::class.java)) }
  else if (response.getStatus() >= 500 && response.getStatus() < 600) { Status5XX(response.getStatus(), response.readEntity(Error::class.java)) }
  else { throw IllegalStateException("Unexpected status code: " + response.getStatus()) } }).onFailure(WebApplicationException::class.java).recoverWithItem({ e: WebApplicationException -> if (e.getResponse().getStatus() == 200) { Status200(e.getResponse().readEntity(object : GenericType<List<Animal>>() {})) }
  else if (e.getResponse().getStatus() >= 400 && e.getResponse().getStatus() < 500) { Status4XX(e.getResponse().getStatus(), e.getResponse().readEntity(Error::class.java)) }
  else if (e.getResponse().getStatus() >= 500 && e.getResponse().getStatus() < 600) { Status5XX(e.getResponse().getStatus(), e.getResponse().readEntity(Error::class.java)) }
  else { throw IllegalStateException("Unexpected status code: " + e.getResponse().getStatus()) } })

  /** List all animals (polymorphic) */
  @GET
  @Path("/")
  @Produces(MediaType.APPLICATION_JSON)
  fun listAnimalsRaw(): Uni<Response>
}

File: testers/openapi/scala/spring/testapi/api/AnimalsApiClient.scala
package testapi.api

import com.fasterxml.jackson.databind.ObjectMapper
import java.lang.Exception
import java.net.URI
import java.net.http.HttpClient
import testapi.model.Animal

/** JDK HTTP Client implementation for AnimalsApi */
class AnimalsApiClient(
    /** JDK HTTP client for making HTTP requests */
    val httpClient: HttpClient,
    /** Base URI for API requests */
    val baseUri: URI,
    /** Jackson ObjectMapper for JSON serialization */
    val objectMapper: ObjectMapper
) extends AnimalsApi {

  /** List all animals (polymorphic) */
  @throws[Exception]
  override def listAnimals: Response2004XX5XX[java.util.List[Animal]] = {
    var request = java.net.http.HttpRequest
      .newBuilder(java.net.URI.create(baseUri.toString() + "/" + "animals"))
      .method("GET", java.net.http.HttpRequest.BodyPublishers.noBody())
      .header("Content-Type", "application/json")
      .header("Accept", "application/json")
      .build()
    var response = httpClient.send(request, java.net.http.HttpResponse.BodyHandlers.ofString())
    var statusCode = response.statusCode()
    if (statusCode == 200) return new testapi.api.Ok(objectMapper.readValue(response.body(), new com.fasterxml.jackson.core.`type`.TypeReference[java.util.List[testapi.model.Animal]] {}))
    else if (statusCode >= 400 && statusCode < 500) return new testapi.api.ClientError4XX(statusCode, objectMapper.readValue(response.body(), classOf[testapi.model.Error]))
    else if (statusCode >= 500 && statusCode < 600) return new testapi.api.ServerError5XX(statusCode, objectMapper.readValue(response.body(), classOf[testapi.model.Error]))
    else throw new java.lang.IllegalStateException(s"Unexpected status code: ${statusCode}")
  }
}
